### PR TITLE
[Ecotone] Extend engine api

### PR DESCRIPTION
**Pull request by ImTei**
This PR mirrors https://github.com/ethereum-optimism/op-geth/pull/214
- Add ParentBeaconBlockRoot into GetPayload response

This PR depends on https://github.com/testinprod-io/erigon-interfaces/pull/11

### CODE REVIEW

**Review by pcw109550 — COMMENTED** [review]
May I ask where is the code for 
> For Bedrock and Canyon parentBeaconBlockRoot MUST be nil and in Ecotone it MUST be set to the parentBeaconBlockRoot from the L1 Origin block of the L2 block.

Ref: https://github.com/ethereum-optimism/optimism/blob/develop/specs/exec-engine.md#engine_getpayloadv3

**Comment by pcw109550** [comment]
Also, please run CI

**Comment by ImTei** [comment]
@pcw109550 Implemented parentBeaconBlockRoot nil check in https://github.com/testinprod-io/op-erigon/pull/126/commits/bae2d5449992610e592d37a344870c9bd17d0258

File: turbo/engineapi/engine_server.go [hunk 1]
```diff
@@ -420,11 +420,20 @@ func (s *EngineServer) getPayload(ctx context.Context, payloadId uint64, version
...
-	return &engine_types.GetPayloadResponse{
+	response := engine_types.GetPayloadResponse{
 		ExecutionPayload: engine_types.ConvertPayloadFromRpc(data.ExecutionPayload),
 		BlockValue:       (*hexutil.Big)(gointerfaces.ConvertH256ToUint256Int(data.BlockValue).ToBig()),
 		BlobsBundle:      engine_types.ConvertBlobsFromRpc(data.BlobsBundle),
-	}, nil
+	}
+	if s.config.IsOptimism() && s.config.IsCancun(ts) && version >= clparams.DenebVersion {
+		if data.ParentBeaconBlockRoot == nil {
+			panic("missing ParentBeaconBlockRoot in Cancun block")
```
Comment (pcw109550) [inline turbo/engineapi/engine_server.go]: nit: `ParentBeaconBlockRoot` is extended in op-stack, so
```suggestion
			panic("missing ParentBeaconBlockRoot in Ecotone block")
```
might be better.

**Review by pcw109550 — APPROVED** [review]

**Pull request merged**